pandas/core/indexing.py

```diff
@@ -663,17 +663,9 @@ def _ensure_listlike_indexer(self, key, axis=None, value=None):
             and not com.is_bool_indexer(key)
             and all(is_hashable(k) for k in key)
         ):
-            for i, k in enumerate(key):
-                if k not in self.obj:
-                    if value is None:
-                        self.obj[k] = np.nan
-                    elif is_array_like(value) and value.ndim == 2:
-                        # GH#37964 have to select columnwise in case of array
-                        self.obj[k] = value[:, i]
-                    elif is_list_like(value):
-                        self.obj[k] = value[i]
-                    else:
-                        self.obj[k] = value
+            keys = self.obj.columns.tolist()
```

Comment:
can you create the keys in the same order whether they are in the obj or not (e.g. combine L666 and 667)

Reply:
do we have a test for this ordering?

Reply:
Union should do the trick.

At least the test added with #37964 covers this